src/FsCheck/Gen.fs

```diff
@@ -181,7 +181,7 @@ module Gen =
         | :? array<_> as arr ->
             elementsArr arr
         | _ ->
-            choose (0, (Seq.length xs)-1) |> map (flip Seq.nth xs)
+            choose (0, (Seq.length xs)-1) |> map (flip Seq.item xs)
```

Comment:
btw i added this change because @baronfel tried to compile ootb with xamarin studio but failed (warning as error?), and the real compiler error of this issue was after these two warnings